### PR TITLE
CORE-6717 Add schemas for hosted identity reconciliation

### PR DESCRIPTION
**Pull request by YashNabar**
Introduces database and Avro schemas to enable persistence and reconciliation of locally-hosted identities.

- Adds cluster-level tables `hosted_identity` and `hosted_identity_session_key_info`
- Extends `HostedIdentityEntry.avsc` to include `version` (compatibility test added)
- Adds config for hosted identity reconciliation

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/p2p/HostedIdentityEntry.avsc [hunk 1]
```diff
@@ -33,6 +33,11 @@
          "type": "array",
          "items": "HostedIdentitySessionKeyAndCert"
        }
-     }
+     },
+    {
+      "name": "version",
+      "type": ["null", "int"],
```
Comment (nikinagy) [inline data/avro-schema/src/main/resources/avro/net/corda/data/p2p/HostedIdentityEntry.avsc:40]: add docs?

**Review by nikinagy — DISMISSED** [review]
LGTM

**Review by owenstanford — DISMISSED** [review]

**Review by driessamyn — DISMISSED** [review]
This cannot go in 5.2 as patch releases must be able to be applied in place (without an upgrade process for DBs etc).

**Comment by nikinagy** [comment]
> This cannot go in 5.2 as patch releases must be able to be applied in place (without an upgrade process for DBs etc).

@driessamyn The main problem is that we were not persisting data in the DB, so in case of kafka data loss we lost everything.
The hosted identities topic was never backed up by any sort of DB entity. Hence we had no DB reconciliation in place.
I don't see how we could fix the SBI issue for clients without making any changes to the DB. Especially, since the main issue is that we are missing persistence for this data.

File: data/config-schema/src/main/resources/net/corda/schema/configuration/reconciliation/1.0/corda.reconciliation.json [hunk 2]
```diff
@@ -61,6 +61,13 @@
       "minimum": 5000,
       "maximum": 2147483647,
       "default": 120000
+    },
```
Comment (simon-johnson-r3) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/reconciliation/1.0/corda.reconciliation.json:64]: Just a heads up, after upgrade it's possible the config read by a worker will not yet have this property with a default propagated to it. The code must be able to handle this property is missing in 5.2.1. See https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4564091182/Making+a+Config+Change for more details.

**Review by simon-johnson-r3 — APPROVED** [review]

**Review by conal-smith — APPROVED** [review]
LGTM

**Pull request merged**